bindings/pyroot/pythonizations/python/ROOT/_pythonization/_roofit/_rooworkspace.py

```diff
@@ -61,8 +61,78 @@ def __setitem__(self, key, value):
+                min_value = None
+                is_constant = False
+                if "value" in value:
+                    val = value["value"]
+                if ("max" in value) and ("min" in value):
+                    max_value = value["max"]
+                    min_value = value["min"]
+                else:
+                    is_constant = True
+                if not (val is None) or not ((max_value is None) and (min_value is None)):
+                    if(val is None):
+                        val = (max_value+min_value)/2
+                    # Create dictionary with domains and parameter points
+                    json_dict = {
+                        "domains": [
+                            {
+                                "axes": [
+                                    {
+                                        "name": key
+                                    }
+                                ],
+                                "name": "default_domain",
+                                "type": "product_domain"
+                            }
+                        ],
+                        "parameter_points": [
+                            {
+                                "name": "default_values",
+                                "parameters": [
+                                    {
+                                        "name": key,
+                                        "value": val
+                                    }
+                                ]
+                            }
+                        ] 
+                    }
+                    if is_constant:
+                        json_dict["parameter_points"][0]["parameters"][0]["const"] = True
+                        json_dict["misc"] = dict()
+                        json_dict["misc"]["ROOT_internal"] = dict()
+                        json_dict["misc"]["ROOT_internal"][key] = dict()
+                        json_dict["misc"]["ROOT_internal"][key]["tags"] = [ "Constant"]
+                    if not ((max_value is None) and (min_value is None)):
+                        json_dict["domains"][0]["axes"][0]["max"] = max_value
+                        json_dict["domains"][0]["axes"][0]["min"] = min_value
+                    json_string = json.dumps(json_dict, separators=(',',':'))
+                    ROOT.RooJSONFactoryWSTool(self).importFromString(json_string, "variable")
+                else:
+                    raise SyntaxError("Invalid Syntax")
+            else:
+                # Imports the object as a distribution if its 'type' ends with '_dist' else imports it as a functions
+                json_string = json.dumps(value, separators=(',', ':'))
+                if(value["type"].endswith("_dist")):
+                    ROOT.RooJSONFactoryWSTool(self).importFromString(json_string, "distribution")
+                else:
+                    ROOT.RooJSONFactoryWSTool(self).importFromString(json_string, "function")
```

Comment:
```suggestion
                ROOT.RooJSONFactoryWSTool(self).importFromString(json_string)
```
Let's try to keep the logic on the Python side minimal, and also not create this overhead by passing extra strings. Why not figure out on the C++ side if this is a distribution or a function?

To do the `endswith` check on the C++ side, you can copy-paste this function inside RooJSONFactoryWSTool.cxx:
https://github.com/root-project/root/blob/master/roofit/hs3/src/JSONFactories_HistFactory.cxx#L79

Or even better, create a new utility header inside `roofit/hs3/src/` that can be used by the RooJSONFactoryWSTool and `JSONFactories_HistFactory`.